code_syntax_layer.go

```diff
@@ -24,12 +24,13 @@ func (l *CodeSyntaxLayer) Clear() {
 }
 
 func (l *CodeSyntaxLayer) UpdateSpans(runeCount int, edits []TextBoxEdit) {
+	if l == nil {
+		return
+	}
 	min := 0
 	max := runeCount
-	for _, e := range edits {
-		if l == nil {
-			continue
-		}
+	for i := range edits {
+		e := &edits[i]
```

Comment:
I looked at the reasoning from the linter, and I disagree with it in this case.  `TextBoxEdit` is a fairly small struct - two `int`s and two `[]rune`s.  The `[]rune` values will only copy the *address* of the contents, not the actual runes themselves - so we're basically copying four `int` values.

As a bonus, we're _probably_ copying them from the heap to the stack, meaning that the rest of the operations with `e` are faster (since access to the heap is much slower than access to the stack).

I'd like to see this (and the other `range` change) reverted back to the old code.

Reply:
I agree, you are right. Updated.